icd/api/pipeline_binary_cache.cpp

```diff
@@ -121,6 +121,11 @@ bool PipelineBinaryCache::IsValidBlob(
     size_t                 dataSize,
     const void*            pData)
 {
+    if ((pData == nullptr) || (pKey == nullptr) || (dataSize < sizeof(PipelineBinaryCachePrivateHeader)))
+    {
+        return false;
+    }
```

Comment:
This should be done without using an early return.

pData can't be null (it's the result of incrementing a pointer in the caller), so an assert is all that is necessary.